apps/mysql.go

```diff
@@ -15,13 +15,75 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
+	"os"
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,hostname_port|file"`
```

Comment:
You can't use the `file` validator because that requires the file to exist at validation time. (Since Ops Agent might start before MySQL, we can't guarantee the MySQL socket exists at config generation time.)

Reply:
I'm not seeing a good replacement for it. and if we remove file, we'll have to remove `hostname_port` or it'll always use that. Any suggestions?

Reply:
@quentinmit So, we can split the field into two separate mutually exclusive fields, one for unix paths that we don't validate, and one for tcp that we validate with `hostname_port`. This would also solve the transport issue comment so we can automatically default to `tcp` or `unix` based on what field is present and require that only one of the two fields are present.

Reply:
How about `hostname_port|startswith=/` and select a transport based on the leading `/`?

Reply:
that works

Reply:
Confirmed that this has been addressed in https://github.com/GoogleCloudPlatform/ops-agent/commit/9858567c7473a8286d3628d9024cb12c5dfe6b26. Resolving this thread.